test/test_stream.py

```diff
@@ -0,0 +1,112 @@
+import expecttest
+import os
+import socketserver
+import tarfile
+import tempfile
+import threading
+import unittest
+
+
+from http.server import SimpleHTTPRequestHandler
+from torch.utils.data.datapipes.iter import StreamReader
+from torchdata.datapipes.iter import HttpReader, IterableWrapper
+from typing import List, Tuple
+
+
+# HTTP server setup
+# Allows the server to start from an arbitrary directory, such as a TempDirectory
+def handler_from(directory):
+    def _init(self, *args, **kwargs):
+        return SimpleHTTPRequestHandler.__init__(
+            self, *args, directory=self.directory, **kwargs
+        )  # type: ignore[call-arg]
+
+    return type(f"HandlerFrom<{directory}>", (SimpleHTTPRequestHandler,), {"__init__": _init, "directory": directory},)
+
+
+class TestStream(expecttest.TestCase):
+    def setUp(self) -> None:
+        # Temporary Directory abd Server Setup
+        print("\nSetup is running...")
+        self.temp_dir = tempfile.TemporaryDirectory()  # noqa: P201
+        self.temp_dir_path = self.temp_dir.name
+        self.port = 8006
+        self.stop_server = False
+        self.server_thread = threading.Thread(
+            target=self.running_server
+        )  # TestStream.start_test_server(self.temp_dir_path, self.port)
```

Comment:
This is not going to work. You need to pass these variables to thread during construction.

Reply:
It seems to work for now since `running_server` has access to `self` variable. But I will keep an eye out to see if there is any bug related to this.

Reply:
But I think it would be better to refactor it and take those as arguments as you suggested.